Fix slow before/after command hook in sync mode

mgrybyk opened the pull request:
## Proposed changes

Fix slow before/after command hook in sync mode. #4133 
Avoid calling before/after hook for all nested commands.

**93% faster now!**

See test used https://github.com/webdriverio/webdriverio/pull/4221#issuecomment-513987108

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

To make this fix working properly we need to merge https://github.com/webdriverio/webdriverio/pull/4229 first and rebase

### Reviewers: @webdriverio/technical-committee

christian-bromann commented on packages/wdio-sync/src/wrapCommand.js:
```diff
@@ -73,3 +40,56 @@ export default function wrapCommand (commandName, fn) {
+}
+
+function shouldSkipHook(fn) {
+    const ignoredFunctions = ['elementErrorHandlerFn', 'wrapCommandFn', 'elementErrorHandlerCallbackFn']
```
Let's move this up to the top of the file

christian-bromann commented on packages/webdriver/src/command.js:
```diff
@@ -88,4 +88,12 @@ export default function (method, endpointUri, commandInfo) {
     }
+
+    /**
+     * skip certain protocol commands as far as these commands are wrapped by WebdriverIO
```
can you elaborate on this?

mgrybyk replied:
I hove the table might help
```
Protocol Command			WebdriverIO Command
-----------------------------------------------
findElement					$
findElements				$$
findElementFromElement		$
findElementsFromElement 	$$
getActiveElement			?
isElementSelected			isSelected
isElementDisplayed			isDisplayed
getElementAttribute			getAttribute
getElementProperty			getProperty
getElementCSSValue			getCSSProperty
getElementText				getText
getElementTagName			getTagName
getElementLocation			getLocation
getElementRect				getSize
isElementEnabled			isEnabled
elementClick				click
elementClear				clearValue
elementSendKeys				addValue
takeElementScreenshot		saveScreenshot

getElementLocationInView	?
getActiveElement			?
elementEquals				?
elementSubmit				?
```

mgrybyk replied:
I might understood the question wrong, if you like we may discuss it offline or you may be more specific.

mgrybyk commented:
> I would like to check out the code by myself and do some more investigations

I'd insist on additional review if possible, thank you

mgrybyk commented:
@christian-bromann can you please take a look again. Simplified the logic, removed function properties.

please DON'T merge until https://github.com/webdriverio/webdriverio/pull/4229 is merged!

mgrybyk commented:
test used for testing
```

describe("perf", () => {
    before(() => {
        browser.url("http://adam.goucher.ca/parkcalc/")
        $("body").waitForExist()
        browser.pause(2000)
    })

    it("before and after command hooks in sync mode performance", () => {
        // Please add before/after command hooks to wdio.conf
        // beforeCommand: function() {},
        // afterCommand: function() {},

        const elems = $$("*") // 50 elements

        const start = Date.now()
        elems.forEach(el => el.getValue())
        console.log(Date.now() - start)

        // tests on my local machine
        // before:  9800
        // after:   630
        //
        // Duration decreased by 93%
        // (before - after) / after * 100 = 93.5%
    })
})
```

mgrybyk commented:
if there are not other comments, let's merge!
